tests/test_browser.py

```diff
@@ -112,6 +112,31 @@ def test__request_file(httpbin):
     assert "multipart/form-data" in response.request.headers["Content-Type"]
 
 
+def test__request_file_none(httpbin):
```

Comment:
Yes, adding a test is great! However, I think this can be accomplished with less code duplication by using the `pytest.mark.parametrize` decorator. For example, see `tests/test_stateful_browser.py` line 430 (`test_follow_link_arg`).

One possible way to use parameters would be to set an `expected_content` (":-)" or "") and `set_value` (a bool to determine whether or not to write the `expected_content` to a temporary file and set the form value for "pic").